ARCHITECTURE.md

```diff
@@ -0,0 +1,152 @@
+![BluepillIcon](doc/img/bluepill_text.png)
+
+![](https://github.com/linkedin/bluepill/workflows/master/badge.svg)
+
+## iOS Simulator Architecture
+
+- App (builds with) -> (Xcode/Bazel) -> `X.app` (packaged into an IPA, an archive of the app or "Bundle")
+- iPhones have ARM processors while Macs have `x86_64` (pre M1 at least)
+- Compilation results in an architecture specific binary (or app binary)
+
+## Bluepill Build How-To
+- `bluepill.sh` builds Bluepill in CI (It has a sample app)
+- Technically the script builds Bluepill & the sample app, and runs the tests
+- `BPSampleApp.app` is the "bundle", an x86_64 bundle but iPhone needs an ARM executable
```

Comment:
X86_64 arch is for iOS simulators running on Intel based Macs. arm64 arch is for iOS device and Apple Silicon based Macs.

For development we usually build only 1 architecture for the target device / simulator. For distribution we would build multi-arch archives to cover the variety of devices.

https://en.wikipedia.org/wiki/Fat_binary for reference.